src/luaState.h

```diff
@@ -51,7 +54,10 @@ class LuaState {
 	static int luaCallableCall(lua_State *state);
 
 	static void luaHook(lua_State *state, lua_Debug *ar);
-
+protected:
+	// Helper functions for recursive indexing
+	void indexForReading(String name); // Puts the object on the stack
+	String indexForWriting(String name); // Puts the table on the stack and gives the last name. (Please make sure the table is not nil.)
```

Comment:
What is the justification for making these protected here? We don't expect any class to derive from this class.

Reply:
I actually have no idea lmao